src/nullablevector.jl

```diff
@@ -311,3 +311,80 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+function Base.promote_rule{T1,T2}(::Type{T1}, ::Type{Nullable{T2}})
```

Comment:
This is not actually needed (and would have to live in Base anyway). Indeed you should never call `promote_rule` directly, but use `promote_type` instead. This already works:
```julia
julia> promote_type(Nullable{Int64}, Int64)
Nullable{Int64}
```